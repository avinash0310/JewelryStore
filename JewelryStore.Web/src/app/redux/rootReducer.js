import { combineReducers } from 'redux';
import loginReducer from '../../features/Login/reducer';
import jewelryStoreReducer from '../../features/JewelryStore/reducer';

const rootReducer = combineReducers({
    loginDetails: loginReducer,
    jewelryStore: jewelryStoreReducer,
});

export default rootReducer;