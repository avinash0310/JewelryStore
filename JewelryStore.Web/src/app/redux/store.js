import { applyMiddleware, createStore } from 'redux';
import rootReducer from './rootReducer';
import { httpMiddleware } from '../middleware/middleware.js';

const store = createStore(rootReducer, applyMiddleware(httpMiddleware));
export default store;