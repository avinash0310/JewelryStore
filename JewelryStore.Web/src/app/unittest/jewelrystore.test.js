import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import Jewelrystore, { JewelryStore } from '../../features/JewelryStore/JewelryStore.jsx';
import rootReducer from '../redux/rootReducer';
import {
    GOLD_PRICE_VALIDATION_MSG, GOLD_WEIGHT_VALIDATION_MSG, DEFAULT_DISCOUNT, DISCOUNT_VALIDATION_MSG,
    PRIVILEGED
} from '../common/constant';

describe('JewelryStore', () => {
    const tempObj = {
        payload: {
            customer: { customerType: PRIVILEGED}
        },
        isFailed: false,
        isLoading: false,
        error: null
    };
    const props = {
        dispatch: jest.fn(),
        loginInfo: tempObj,
        jewelryStore: tempObj,
        history: []
    };

    it("should render initial layout", () => {
        // when
        let store = createStore(rootReducer, applyMiddleware())
        let component = shallow(<Provider store={store}><Jewelrystore /></Provider>);

        // then
        expect(component.getElements()).toMatchSnapshot();
    });

    it('gold price input check', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        wrapper.find('input[type="number"]').first().simulate('change', { target: { name: 'goldPrice', value: '123' } });

        // then
        expect(wrapper.state('goldPrice')).toEqual('123');
    });

    it('gold weight input check', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        wrapper.find('input[type="number"]').at(1).simulate('change', { target: { name: 'goldWeight', value: '123' } });

        // then
        expect(wrapper.state('goldWeight')).toEqual('123');
    });

    it('total prize should be empty', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);

        // then
        expect(wrapper.state('totalPrice')).toEqual("");
    });

    it('total prize should not be set', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        wrapper.find('input[type="text"]').simulate('change', { target: { name: 'totalPrice', value: '123' } });

        // then
        expect(wrapper.state('totalPrice')).toEqual("");
    });

    it('discount input check', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        wrapper.find('input[type="number"]').at(2).simulate('change', { target: { name: 'discount', value: '12' } });

        // then
        expect(wrapper.state('discount')).toEqual('12');
    });

    it('discount input check', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);

        // then
        expect(wrapper.state('discount')).toEqual(DEFAULT_DISCOUNT);
    });

    it('calculate check with wrong gold price', () => {
        // when
        window.alert = jest.fn();
        let wrapper = shallow(<JewelryStore {...props} />);
        wrapper.find('input[type="number"]').first().simulate('change', { target: { name: 'goldPrice', value: '0' } });
        wrapper.find('button').first().simulate('click', {
            preventDefault: () => {
            }
        });

        // then
        expect(window.alert).toHaveBeenCalledWith(GOLD_PRICE_VALIDATION_MSG);
        expect(wrapper.state('isDisabled')).toBe(false);
    });

    it('calculate check with wrong gold weight', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        window.alert = jest.fn();
        wrapper.find('input[type="number"]').at(1).simulate('change', { target: { name: 'goldWeight', value: '0' } });
        wrapper.find('input[type="number"]').first().simulate('change', { target: { name: 'goldPrice', value: '123' } });
        wrapper.find('button').first().simulate('click', {
            preventDefault: () => {
            }
        });

        // then
        expect(window.alert).toHaveBeenCalledWith(GOLD_WEIGHT_VALIDATION_MSG);
        expect(wrapper.state('isDisabled')).toBe(false);
    });

    it('calculate check with wrong discount', () => {
        // when
        let wrapper = shallow(<JewelryStore {...props} />);
        window.alert = jest.fn();
        wrapper.find('input[type="number"]').at(1).simulate('change', { target: { name: 'goldWeight', value: '123' } });
        wrapper.find('input[type="number"]').at(2).simulate('change', { target: { name: 'discount', value: '123' } });
        wrapper.find('input[type="number"]').first().simulate('change', { target: { name: 'goldPrice', value: '123' } });
        wrapper.find('button').first().simulate('click', {
            preventDefault: () => {
            }
        });

        // then
        expect(window.alert).toHaveBeenCalledWith(DISCOUNT_VALIDATION_MSG);
        expect(wrapper.state('isDisabled')).toBe(false);
    });
});