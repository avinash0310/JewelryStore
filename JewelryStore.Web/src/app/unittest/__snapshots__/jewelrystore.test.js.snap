// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JewelryStore should render initial layout 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Memo(Connect(JewelryStore)) />
  </Context.Provider>,
]
`;
